accounts/abi/bind/backends/simulated.go

```diff
@@ -109,12 +112,28 @@ func (b *SimulatedBackend) Commit() {
 	b.mu.Lock()
 	defer b.mu.Unlock()
 
-	if _, err := b.blockchain.InsertChain([]*types.Block{b.pendingBlock}); err != nil {
+	// Get the last block
+	parentBlock := b.blockchain.GetBlockByHash(b.pendingBlock.ParentHash())
+	var remainingTx types.Transactions
+
+	blocks, _ := core.GenerateChain(b.config, parentBlock, ethash.NewFaker(), b.database, 1,
+		func(number int, block *core.BlockGen) {
+			for _, tx := range append(b.pendingBlock.Transactions(), b.stuckTransactions...) {
+				if b.marketGasPrice == nil || b.marketGasPrice.Cmp(tx.GasPrice()) <= 0 {
+					block.AddTxWithChain(b.blockchain, tx)
+				} else {
+					remainingTx = append(remainingTx, tx)
+				}
+			}
+		})
+
+	if _, err := b.blockchain.InsertChain([]*types.Block{blocks[0]}); err != nil {
 		panic(err) // This cannot happen unless the simulator is wrong, fail in that case
 	}
 	// Using the last inserted block here makes it possible to build on a side
 	// chain after a fork.
-	b.rollback(b.pendingBlock)
+	b.stuckTransactions = remainingTx
```

Comment:
can we instead just put them back on the pending block?

Reply:
we cannot put them to the pending block, the txes in the pending block determine whats the next nonce available. if they are stuck, they should not be in pending block. because that means we cannot send the transaction with the same nonce. it will be invalid/nonce too low.